libraries/AP_HAL_Linux/GPIO_Sysfs.cpp

```diff
@@ -41,20 +53,18 @@ DigitalSource_Sysfs::~DigitalSource_Sysfs()
 uint8_t DigitalSource_Sysfs::read()
 {
     char char_value;
-    int r = ::pread(_value_fd, &char_value, 1, 0);
-    if (r < 0) {
-        hal.console->printf("warning: unable to read GPIO value for pin %u\n",
+    if (::pread(_value_fd, &char_value, 1, 0) < 0) {
+        hal.console->printf("DigitalSource_Sysfs: Unable to read pin %u value.\n",
```

Comment:
pread() is the equivalent to lseek() + read().

If you mix reads and writes or are using interrupts (not in this case) you may need to "rewind" the fd offset in order to read it correctly. See the sysfs doc:
```
        "value" ... reads as either 0 (low) or 1 (high). If the GPIO                                                                                                                                                                                          
                is configured as an output, this value may be written;                                                                                                                                                                                        
                any nonzero value is treated as high.                                                                                                                                                                                                         

                If the pin can be configured as interrupt-generating interrupt                                                                                                                                                                                
                and if it has been configured to generate interrupts (see the                                                                                                                                                                                 
                description of "edge"), you can poll(2) on that file and                                                                                                                                                                                      
                poll(2) will return whenever the interrupt was triggered. If                                                                                                                                                                                  
                you use poll(2), set the events POLLPRI and POLLERR. If you                                                                                                                                                                                   
                use select(2), set the file descriptor in exceptfds. After                                                                                                                                                                                    
                poll(2) returns, either lseek(2) to the beginning of the sysfs                                                                                                                                                                                
                file and read the new value or close the file and re-open it                                                                                                                                                                                  
                to read the value.                                                                                                                                                                                                                           
```